<a href="https://colab.research.google.com/github/JosenildoJunior/StatPyDataScience/blob/main/Desafio_Ifood.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Análise exploratória em dados do Ifood**

O conjunto de dados é composto por clientes da empresa Ifood com dados sobre:

- Perfis de clientes
- Preferências do produto
- Sucessos/fracassos da campanha
- Desempenho do canal


**O objetivo é realizar uma análise exploratória desses dados.**

## **Importando os dados**

Nessa parte iremos importar todas as bibliotecas que iremos precisar para realizar nossa analise e também os dados, dito isso vamos seguir para a importação das bibliotecas e posteriormente dos dados

In [1]:
# Manipulação de dados
import pandas as pd

# Algébra linear
import numpy as np

# Visualização de dados
import matplotlib.pyplot as plt
import seaborn as sns

# Funções estatísticas
import statistics
import scipy.stats

Com as bibliotecas importadas podemos seguir para a importação dos dados propriamente dito

In [2]:
# Acesso ao drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# Importando o dataset
df = pd.read_csv('/content/drive/MyDrive/Estatística para ciência de dados/mkt_data.csv')

# Observando os primeiro registros
df.head()

Unnamed: 0   Income  Kidhome  Teenhome  Recency  MntWines  MntFruits  \
0           0  58138.0        0         0       58       635         88   
1           1  46344.0        1         1       38        11          1   
2           2  71613.0        0         0       26       426         49   
3           3  26646.0        1         0       26        11          4   
4           4  58293.0        1         0       94       173         43   

   MntMeatProducts  MntFishProducts  MntSweetProducts  ...  \
0              546              172                88  ...   
1                6                2                 1  ...   
2              127              111                21  ...   
3               20               10                 3  ...   
4              118               46                27  ...   

   education_Graduation  education_Master  education_PhD  MntTotal  \
0                   3.0               NaN            NaN      1529   
1                   3.0               NaN            NaN        21   
2                   3.0               NaN            NaN       734   
3                   3.0               NaN            NaN        48   
4                   NaN               NaN            5.0       407   

   MntRegularProds  AcceptedCmpOverall  marital_status  education_level  kids  \
0             1441                   0          Single       Graduation     0   
1               15                   0          Single       Graduation     2   
2              692                   0        Together       Graduation     0   
3               43                   0        Together       Graduation     1   
4              392                   0         Married              PhD     1   

   expenses  
0      1529  
1        21  
2       734  
3        48  
4       407  

[5 rows x 44 columns]

Agora podemos partir para a proxima etapa

## **Análise inicial**

Nessa parte do desafio responderemos algumas perguntas com o intuito de enterdermos um pouco mais de como esses dados se comportam, para isso responderemos as seguintes perguntas:

- Quantos dados temos? Linhas e colunas
- Quais são as colunas numéricas?
- Temos duplicados na nossa base? Se tivermos, retire-os
- Temos dados nulos nessa base? Será que eles indicam algo? O que fazer com eles?
- Qual é a média, mediana, 25 percentil, 75 percentil, mínimo e máximo de cada uma das colunas numéricas?

Agora que já sabemos as perguntas, vamos partir para a resolução das mesmas.

### **Quantos dados temos? Linhas e colunas**

In [4]:
# Observando as dimensões dos dados
df.shape

(2205, 44)

Dessa forma, podemos observar que temos 2205 linhas e 44 colunas nessa base de dados.

### **Quais são as colunas numéricas?**

In [5]:
# Observando o tipo de dado
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2205 entries, 0 to 2204
Data columns (total 44 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            2205 non-null   int64  
 1   Income                2205 non-null   float64
 2   Kidhome               2205 non-null   int64  
 3   Teenhome              2205 non-null   int64  
 4   Recency               2205 non-null   int64  
 5   MntWines              2205 non-null   int64  
 6   MntFruits             2205 non-null   int64  
 7   MntMeatProducts       2205 non-null   int64  
 8   MntFishProducts       2205 non-null   int64  
 9   MntSweetProducts      2205 non-null   int64  
 10  MntGoldProds          2205 non-null   int64  
 11  NumDealsPurchases     2205 non-null   int64  
 12  NumWebPurchases       2205 non-null   int64  
 13  NumCatalogPurchases   2205 non-null   int64  
 14  NumStorePurchases     2205 non-null   int64  
 15  NumWebVisitsMonth    

Dessa forma, podemos observar o tipo de dado presente em cada uma das colunas. Temos 2 variáveis categóricas e 42 colunas numéricas, sendo elas:

**Variáveis categóricas:**
 - marital_status        
 - education_level

**Variáveis numéricas:**
- Income
- Kidhome
- Teenhome              
- Recency               
- MntWines              
- MntFruits             
- MntMeatProducts       
- MntFishProducts     
-  MntSweetProducts       
- MntGoldProds          
- NumDealsPurchases
- NumWebPurchases       
- NumCatalogPurchases    
- NumStorePurchases     
- NumWebVisitsMonth    
- AcceptedCmp3        
- AcceptedCmp4         
- AcceptedCmp5          
- AcceptedCmp1          
- AcceptedCmp2          
- Complain             
- Z_CostContact        
-  Z_Revenue             
-  Response              
- Age                   
- Customer_Days       
- marital_Divorced     
- marital_Married      
- marital_Single        
- marital_Together    
- marital_Widow         
- education_2n Cycle    
- education_Basic       
- education_Graduation  
- education_Master     
-  education_PhD         
-  MntTotal              
-  MntRegularProds       
-  AcceptedCmpOverall   
-  kids                
-  expenses             




### **Temos duplicados na nossa base? Se tivermos, retire-os**

In [9]:
# Verificando o valores duplicados
df.duplicated().sum()

0

Podemos observar que nesta base não existem valores duplicados. Sendo assim, podemos seguir adiante.